src/Microsoft.Extensions.Http/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs

```diff
@@ -69,5 +82,116 @@ public static IServiceCollection AddHttpClient(this IServiceCollection services,
 
             return services;
         }
+
+        /// <summary>
+        /// Adds the <see cref="IHttpClientFactory"/> and related services to the <see cref="IServiceCollection"/> and configures
+        /// all <see cref="HttpClient"/> instances produced by the <see cref="IHttpClientFactory"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        /// <param name="configureClient">
+        /// A delegate that is used to configure all <see cref="HttpClient"/> instances produced by the
+        /// <see cref="IHttpClientFactory"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/>.</returns>
+        /// <remarks>
+        /// <para>
+        /// The <paramref name="configureClient"/> delegate will be used to configure all <see cref="HttpClient"/> instances
+        /// produced by the <see cref="IHttpClientFactory"/>.
+        /// </para>
+        /// </remarks>
+        public static IServiceCollection AddGlobalHttpClient(this IServiceCollection services, Action<HttpClient> configureClient)
```

Comment:
I'm going with the approach right now of having a series of extensions on the `IServiceCollection` directly instead of an `IHttpClientFactoryBuilder` because I want to see if it's necessary.

 If we feel like this isn't scaling as we add more functionality we will pivot.

Reply:
Given that most 3rd-party extensions will hang off the `HttpMessageHandlerBuilder` I think that's probably OK.

Reply:
I dislike polluting the top level IServiceCollection with features of subcomponents. I'd need to see a few extension methods before I judge though. I think having a more specific type is better for scoping.